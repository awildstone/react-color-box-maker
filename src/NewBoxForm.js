import React, { useState } from 'react';
import { nanoid } from 'nanoid';

const NewBoxForm = ({ addBox }) => {
    const INITIAL_STATE = {
        bgColor: 'black',
        width: '',
        height: ''
    };

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(formData => ({...formData, [name]: value}));
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();
        addBox({ ...formData, id: nanoid() });
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="bgColor">Choose a color</label>
            <input
                id="bgColor"
                type="color"
                name="bgColor"
                value={formData.bgColor}
                onChange={handleChange}
            />
            <label htmlFor="width">Width</label>
            <input
                id="width"
                type="text"
                name="width"
                value={formData.width}
                onChange={handleChange}
            />
            <label htmlFor="height">Height</label>
            <input
                id="height"
                type="text"
                name="height"
                value={formData.height}
                onChange={handleChange}
            />
            <button>Add</button>
        </form>
    )
}

export default NewBoxForm;