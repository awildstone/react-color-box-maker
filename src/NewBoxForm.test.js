import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

it('renders a NewBoxForm without crashing', () => {
    render(<NewBoxForm />);
});

it('matches a NewBoxForm snapshot', () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});