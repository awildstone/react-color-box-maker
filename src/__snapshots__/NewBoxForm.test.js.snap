// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a NewBoxForm snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="bgColor"
    >
      Choose a color
    </label>
    <input
      id="bgColor"
      name="bgColor"
      type="color"
      value="black"
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
